core/app/models/spree/product/scopes.rb

```diff
@@ -77,8 +77,8 @@ def self.property_conditions(property)
     #   SELECT COUNT(*) ...
     add_search_scope :in_taxon do |taxon|
       includes(:classifications).
-      where("spree_products_taxons.taxon_id" => taxon.self_and_descendants.pluck(:id)).
-      order("spree_products_taxons.position ASC")
+      where("spree_classifications.taxon_id" => taxon.self_and_descendants.pluck(:id)).
```

Comment:
Give this a try:

```Ruby
      add_search_scope :in_taxon do |taxon|
        joins(:classifications).
        where(Spree::Classification.table_name => {
          taxon_id: taxon.self_and_descendants
        }).
        order(Spree::Classification.arel_table[:position].asc)
      end
```

Comparison:

Before:
```
[1] pry(main)> Spree::Product.in_taxon Spree::Taxon.find_by!(permalink: "categories/bags")
  Spree::Taxon Load (0.2ms)  SELECT  "spree_taxons".* FROM "spree_taxons" WHERE "spree_taxons"."permalink" = ? LIMIT 1  [["permalink", "categories/bags"]]
   (0.2ms)  SELECT "spree_taxons"."id" FROM "spree_taxons" WHERE ("spree_taxons"."lft" >= 2) AND ("spree_taxons"."lft" < 3)  ORDER BY "spree_taxons"."lft"
=>   SQL (0.3ms)  SELECT "spree_products"."id" AS t0_r0, "spree_products"."name" AS t0_r1, "spree_products"."description" AS t0_r2, "spree_products"."available_on" AS t0_r3, "spree_products"."deleted_at" AS t0_r4, "spree_products"."slug" AS t0_r5, "spree_products"."meta_description" AS t0_r6, "spree_products"."meta_keywords" AS t0_r7, "spree_products"."tax_category_id" AS t0_r8, "spree_products"."shipping_category_id" AS t0_r9, "spree_products"."created_at" AS t0_r10, "spree_products"."updated_at" AS t0_r11, "spree_products"."promotionable" AS t0_r12, "spree_products"."meta_title" AS t0_r13, "spree_classifications"."product_id" AS t1_r0, "spree_classifications"."taxon_id" AS t1_r1, "spree_classifications"."id" AS t1_r2, "spree_classifications"."position" AS t1_r3 FROM "spree_products" LEFT OUTER JOIN "spree_classifications" ON "spree_classifications"."product_id" = "spree_products"."id" WHERE "spree_products"."deleted_at" IS NULL AND "spree_classifications"."taxon_id" = 3  ORDER BY spree_classifications.position ASC
```

After:
```
[2] pry(main)> Spree::Product.in_taxon Spree::Taxon.find_by!(permalink: "categories/bags")
  Spree::Taxon Load (0.4ms)  SELECT  "spree_taxons".* FROM "spree_taxons" WHERE "spree_taxons"."permalink" = ? LIMIT 1  [["permalink", "categories/bags"]]
=>   Spree::Product Load (0.4ms)  SELECT "spree_products".* FROM "spree_products" INNER JOIN "spree_classifications" ON "spree_classifications"."product_id" = "spree_products"."id" WHERE "spree_products"."deleted_at" IS NULL AND "spree_classifications"."taxon_id" IN (SELECT "spree_taxons"."id" FROM "spree_taxons" WHERE ("spree_taxons"."lft" >= 2) AND ("spree_taxons"."lft" < 3)  ORDER BY "spree_taxons"."lft")  ORDER BY "spree_classifications"."position" ASC
```

Query is simpler and easier to understand, plus we do it all in one shot query (with a subselect) rather than two. It functions a bit differently internally but should produce the intended results.